packages/math-input/src/components/keypad/mobile-keypad.tsx

```diff
@@ -158,12 +163,18 @@ class MobileKeypad extends React.Component<Props, State> implements KeypadAPI {
 
     render(): React.ReactNode {
         const {style} = this.props;
-        const {active, containerWidth, cursor, keypadConfig} = this.state;
+        const {active, hasBeenActivated, containerWidth, cursor, keypadConfig} =
+            this.state;
 
         const containerStyle = [
             // internal styles
             styles.keypadContainer,
             active && styles.activeKeypadContainer,
+            // If the keypad is yet to have ever been activated, we keep it invisible
+            // so as to avoid, e.g., the keypad flashing at the bottom of the page
+            // during the initial render.
+            // Done inline since stylesheets might not be loaded yet.
+            !hasBeenActivated && {visibility: "hidden"},
```

Comment:
I don't understand how this fixes things.  The base `styles.keypadContainer` already has `visibility:hidden` on it. So if `active == false`, shouldn't that be the `visibility` style we have already?  

<img width="361" alt="image" src="https://github.com/Khan/perseus/assets/77138/69260464-96bb-4b87-914a-1906383b9698">

Reply:
Sorry, I pushed this to trigger a snapshot to test in Webapp but I didn't originally mark this as a WIP.

It needs to be added to `dynamicStyle` so that it happens before stylesheets are applied.